b2luigi/batch/processes/htcondor.py

```diff
@@ -0,0 +1,225 @@
+            if status_dict:
+                dict_list.append(json.loads(status_dict))
+            else:
+                continue
+
+        if job_id:
+            if job_id not in [status_dict["ClusterId"] for status_dict in dict_list]:
+                history_cmd.append(str(job_id))
+                output = subprocess.check_output(history_cmd)
+                output = output.decode()
+                output = output.lstrip("[\n").rstrip("\n]\n").split("\n,\n")
+                if output:
+                    for status_dict in output:
+                        dict_list.append(json.loads(status_dict))
+
+        for status_dict in dict_list:
+            self[status_dict["ClusterId"]] = status_dict["JobStatus"]  # int -> int
+
+
+class HTCondorJobStatus(enum.IntEnum):
+    idle = 1
+    running = 2
+    removed = 3
+    completed = 4
+    held = 5
+    submission_err = 6
+
+
+_batch_job_status_cache = HTCondorJobStatusCache()
+
+
+class HTCondorProcess(BatchProcess):
+    """
+    Reference implementation of the batch process for a HTCondor batch system.
+
+    We assume that the batch system shares a file system with the submission node you
+    are currently working on (or at least the current folder and the result/log directory
+    are also available there with the same path).
+    The environment has to be set up on its own new for each job. For that, a environment
+    setup script has to be provided in the ``settings.json`` file.
+    If no own python cmd is specified, the task is executed with the current ``python3``
+    available after the environment is setup.
+    General settings (may be dependend on your HTCondor setup) that affect all jobs (tasks)
+    can be specified in the ``settings.json`` by adding a ``htcondor_settings`` entry.
+    Job specific settings, e.g. number of cpus or required memory can be specified by adding
+    a ``htcondor_settings`` attribute to the task. It's value has to be a dictionary containig
+    also HTCondor settings as key/value pairs.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._batch_job_id = None
+
+    def get_job_status(self):
+
+        assert self._batch_job_id
+
+        try:
+            job_status = _batch_job_status_cache[self._batch_job_id]
+        except KeyError:
+            return JobStatus.aborted
+        if job_status == HTCondorJobStatus.completed:
+            return JobStatus.successful
+        elif job_status == HTCondorJobStatus.idle:
+            return JobStatus.running
+        elif job_status == HTCondorJobStatus.running:
+            return JobStatus.running
+        elif job_status == HTCondorJobStatus.removed:
+            return JobStatus.aborted
+        elif job_status == HTCondorJobStatus.held:
+            return JobStatus.aborted
+        elif job_status == HTCondorJobStatus.submission_err:
+            return JobStatus.aborted
```

Comment:
Maybe you could refactor this into
```
if job_status in [HTCondorJobStatus.completed]:
     return JobStatus.successful
elif job_status in [HTCondorJobStatus.idle, HTCondorJobStatus.running]:
     return JobStatus.running
elif job_status in [HTCondorJobStatus.removed, HTCondorJobStatus.held, HTCondorJobStatus.submission_err]:
    return JobStatus.aborted
```

Question (because I do not know it better): is it worth to have an Exception thrown on `submission_err`?

Reply:
Is refactored